octodns/processor/filter.py

```diff
@@ -215,6 +215,111 @@ def __init__(self, name, rejectlist):
         super().__init__(name, rejectlist)
 
 
+class _ValueBaseFilter(_FilterProcessor):
+    def __init__(self, name, _list, **kwargs):
+        super().__init__(name, **kwargs)
+        exact = set()
+        regex = []
+        for pattern in _list:
+            if pattern.startswith('/'):
+                regex.append(re_compile(pattern[1:-1]))
+            else:
+                exact.add(pattern)
+        self.exact = exact
+        self.regex = regex
+
+    def _process(self, zone, *args, **kwargs):
+        for record in zone.records:
+            values = []
+            if hasattr(record, 'values'):
+                values = [str(value) for value in record.values]
```

Comment:
It seems a bit funky to string-ify complex values here, e.g. a `SRV` or something like that and it's not 100% clear what this would do. After looking at the code it seems like most (maybe all) the value types have `__repr__` methods that this would trigger and thus this would hit those. 

It may be preferable to be explicit here and use the `rdata_text` property/method and call out that's what happens in the documentation. That way people could write matchers (regex or exact) to match on them and we'd be explicit in what's going to be used for the matching. 

https://github.com/octodns/octodns/blob/30f8b4f5f5e9b613c1282615d645e0e2b3f71015/octodns/record/srv.py#L137-L138